specs/src/specs/shares.md

```diff
@@ -0,0 +1,90 @@
+![fig: share continuation](./figures/share_continuation.svg)
+
+Since blob data that exceeds [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_SIZE`](./consensus.md#constants)`-`[`SHARE_INFO_BYTES`](./consensus.md#constants) `-` [`SEQUENCE_BYTES`](./consensus.md#constants) bytes will span more than one share, developers may choose to encode additional metadata in their raw blob data prior to inclusion in a Celestia block.
+
+## Transaction Shares
+
+In order for clients to parse shares in the middle of a sequence without downloading antecedent shares, Celestia encodes additional metadata in the shares associated with reserved namespaces. At the time of writing this only applies to the [`TRANSACTION_NAMESPACE`](./consensus.md#reserved-namespaces) and [`PAY_FOR_BLOB_NAMESPACE`](./consensus.md#reserved-namespaces). This share structure is often reffered to as "compact shares" to differentiate from the share structure defined above (a.k.a "sparse shares").
+
+For shares **with a reserved namespace through [`NAMESPACE_ID_MAX_RESERVED`](./consensus.md#constants)**:
+
+- The first [`NAMESPACE_SIZE`](./consensus.md#constants) of a share's raw data is the namespace of that share.
+- The next [`SHARE_INFO_BYTES`](./consensus.md#constants) bytes are for share information with the following structure:
+  - The first 7 bits represent the share version in big endian form (initially, this will be `0000000` for version `0`);
+  - The last bit is a sequence start indicator. The indicator is `1` if the share is the first share of a sequence or `0` if the share is a continuation share of a sequence.
+- If this is the first share of a sequence the next [`SEQUENCE_BYTES`](./consensus.md#constants) contain a big endian `uint32` that represents the length of the sequence that follows in bytes.
+- The next [`SHARE_RESERVED_BYTES`](./consensus.md#constants) bytes are the starting byte of the length of the [canonically serialized](./consensus.md#serialization) first request that starts in the share, or `0` if there is none, as an unsigned [varint](https://developers.google.com/protocol-buffers/docs/encoding).
+- The remaining [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_SIZE`](./consensus.md#constants)`-`[`SHARE_INFO_BYTES`](./consensus.md#constants) `-` [`SEQUENCE_BYTES`](./consensus.md#constants) `-` [`SHARE_RESERVED_BYTES`](./consensus.md#constants) bytes (if first share) or [`SHARE_SIZE`](./consensus.md#constants)`-`[`NAMESPACE_SIZE`](./consensus.md#constants)`-`[`SHARE_INFO_BYTES`](./consensus.md#constants)`-`[`SHARE_RESERVED_BYTES`](./consensus.md#constants) bytes (if continuation share) are transaction or PayForBlob transaction data. Each transaction or PayForBlob transaction is prefixed with a [varint](https://developers.google.com/protocol-buffers/docs/encoding) of the length of that unit.
+- If there is insufficient transaction or PayForBlob transaction data to fill the share, the remaining bytes are filled with `0`.
+
+First share in a sequence:
```

Comment:
Sure! Do you mind elaborating on what you mean by numerical example? Something like this:

For example, a blob of 1600 bytes would be split into:

```ascii
| share 1 | share 2 | share 3 | share 4 |
| start   | cont.   | cont.   | cont.   |
```

Reply:
What I meant is basically adding some textual elaboration about what is already in the figures, something like below (the example is not completed, I just sketched the idea):  
Suppose there are only 3 transactions tx1 tx2 and tx3 with X, Y, Z sizes (X,Y,Z to be the exact byte sizes according to the figure) in the `TRANSACTION_NAMESPACE`.  These transactions will span two shares  (based on the figures)  with the following values:
Share1:
namespace version: 0
namespace ID: TRANSACTION_NAMESPACE
share version: 0000000
sequence indicator: 1
sequence length: X+Y+Z (the exact sizes though)
reserved byte: 38 (with an explanation why 38, because tx1 starts at byte index 38)
... plus tx1 and tx2 (up to the middle as it hits the share boundary)
Share1 would be serialized as:
`0<namespace verion>||TRANSACTION_NAMESPACE<namespace ID>|| 0000000<share version>||1<sequence indicator>||X+Y+X<sequence length>||38<reserved byte>||...`

Reply:
Created https://github.com/celestiaorg/celestia-app/issues/1941